drivers/flash/flash_stm32g4x.c

```diff
@@ -99,10 +102,26 @@ static int erase_page(struct device *dev, unsigned int page)
 		return rc;
 	}
 
+	if (page > 127) {
```

Comment:
Please put this under `#ifdef FLASH_OPTR_DBANK`

Reply:
The register has only 6 bits. Numbers > 127 are never allowed.

Reply:
Done